pandas/_libs/src/vendored/ujson/python/objToJSON.c

```diff
@@ -373,6 +373,31 @@ static char *PyTimeToJSON(JSOBJ _obj, JSONTypeContext *tc, size_t *outLen) {
   return outValue;
 }
 
+static char *PyDecimalToUTF8Callback(JSOBJ _obj, JSONTypeContext *tc,
+                                     size_t *len) {
+  PyObject *obj = (PyObject *)_obj;
+  PyObject *str = PyObject_Str(obj);
```

Comment:
Looks like we don't have any `Py_DECREF(str)` calls here - make sure to add them when the lifetime of this object is complete, otherwise this will leak resources